colcon_core/which_python.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2019 Dan Rose, no rights reserved
+# Licensed under the Apache License, Version 2.0
+
+from colcon_core.logging import colcon_logger
+from colcon_core.environment_variable import EnvironmentVariable
+import os
+import shutil
+
+logger = colcon_logger.getChild(__name__)
+
+"""Environment variable to set the python executable"""
+PYTHON_EXECUTABLE_ENVIRONMENT_VARIABLE = EnvironmentVariable(
+    'COLCON_PYTHON_EXECUTABLE',
+    'Set the python executable (default: `which python3`)')
```

Comment:
The question here is what should the default value be if the env var is not set. Based on the discussion in #183 it should be `sys.executable` since we can't make the assumption here that it should be `python3`. E.g. if the user installed the package with `python3.6` because `python3` on his system is too old (<3.5) then this would break for them.